lib/cloudregister/registerutils.py

```diff
@@ -1635,3 +1649,29 @@ def __replace_url_target(config_files, new_smt):
                     current_service_server,
                     new_smt.get_FQDN())
                 )
+
+
+# -----------------------------------------------------------------------------
+def __connection_check(smt_ip_addr, propagate=False):
+    if isinstance(smt_ip_addr, ipaddress.IPv6Address):
+        # Per rfc3986 IPv6 addresses in a URI are enclosed in []
+        smt_ip_addr= "[{}]".format(smt_ip_addr)
+
+    protocol = 'http'  # Default for backward compatibility
+    if https_only(get_config()):
+        protocol = 'https'
+    try:
+        cert_res = requests.get(
+            '%s://%s/smt.crt' % (protocol, smt_ip_addr),
+            timeout=3,
+            verify=False
+        )
+    except requests.exceptions.ConnectionError as err:
+        if propagate:
+            raise
+        logging.error(err)
+    except Exception:
+        logging.error('Update server not reachable for %s' % smt_ip_addr)
+        return False
+
```

Comment:
The intend is to determine if the client/instance has IPv4|IPv6 enabled, `is_responsive` could be extended for that purpose if we added the exception handling (with the proper messaging) of this PR
I did not do that so `is_responsive` remains as is and no other code updates/calls to `is_responsive` needed

Reply:
Then check if the client has IPv6 or IPv4 enabled don't implement some detour way of making that check. There are system tools to do such checks, it is not python-requests. Also the intend stated here "The intend is to determine if the client/instance has IPv4|IPv6 enabled" does not match with the descriptive comment "When using '--force-new --smt-ip ...' options, validate that the instance can use the --smt-ip value in order to save debugging or unnecessary errors". Well if one squints really hard then maybe.

The code tries to get the cert from the update server while `is_responsive` uses the designated API of the update server. So ultimately you are achieving the same effect as `is_responsive`

Reply:
I get that there are system tools to do such checks, [this way](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/lib/cloudregister/registerutils.py#L268) does not work, I had dismissed the checking of the metadata
for that info, as that would mean a different way per CSP

I followed the pattern of `is_responsive` or [has_ipv6_access](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/lib/cloudregister/registerutils.py#L877) which is making a query and check/handle the response

I can explore other system options instead of querying the RMT update server, no problem